Source/lua/randomat2/events/gravity.lua

```diff
@@ -0,0 +1,77 @@
+local EVENT = {}
+
+CreateConVar("randomat_gravity_timer", 30, {FCVAR_ARCHIVE, FCVAR_NOTIFY}, "Timer (def. 30)", 5,90)
+CreateConVar("randomat_gravity_duration", 3, {FCVAR_ARCHIVE, FCVAR_NOTIFY}, "Duration (def. 3)", 1,15)
+CreateConVar("randomat_gravity_minimum", 70, {FCVAR_ARCHIVE, FCVAR_NOTIFY}, "The minimum gravity (def. 70)",10,500)
+CreateConVar("randomat_gravity_maximum", 2000, {FCVAR_ARCHIVE, FCVAR_NOTIFY}, "The minimum gravity (def. 70)",700,4000)
+
+EVENT.Title = "I don't think you realise the gravity of the situation."
+EVENT.id = "gravity"
+EVENT.AltTitle = "Gravity Changer"
+
+local defaultGravity = 600
+
+local function SetGravity(increase)
+    if increase then
+        RunConsoleCommand("sv_gravity", GetConVar("randomat_gravity_maximum"):GetFloat())
+    else
+        RunConsoleCommand("sv_gravity", GetConVar("randomat_gravity_minimum"):GetFloat())
+    end
+    return not increase
+end
+
+function EVENT:Begin()
+    defaultGravity = GetConVar("sv_gravity"):GetFloat()
+    if SERVER then
+        SetGravity(true)
+        timer.Simple(duration, function()
+            RunConsoleCommand("sv_gravity", defaultGravity)
+        end)
+    end
+    self:Timer()
+end
+
+function EVENT:Timer()
+    local delay = GetConVar("randomat_gravity_timer"):GetFloat()
+    local duration = GetConVar("randomat_gravity_duration"):GetFloat()
+    local increase = false -- false for min, true for max
+    
+    timer.Create("RandomatGravityChange", delay,0, function()
+        if SERVER then
+            increase = SetGravity(increase)
+            timer.Simple(duration, function()
+                RunConsoleCommand("sv_gravity", defaultGravity)
+            end)
+        end
+        
+    end)
+end
+
+function EVENT:End()
+    self:CleanUpHooks()
```

Comment:
This isn't needed unless you use self:AddHook in the event. I know you copied it from the Moon Gravity event but I'm removing it from there too =)

Reply:
Removed